--- synaptic/common/indexcopy.cc
+++ synaptic/common/indexcopy.cc
@@ -10,6 +10,7 @@
    ##################################################################### */
                                                                         /*}}} */
 // Include Files                                                        /*{{{*/
+#include "config.h"
 #include "indexcopy.h"
 #include "i18n.h"
 
--- synaptic/common/pkg_acqfile.cc
+++ synaptic/common/pkg_acqfile.cc
@@ -21,9 +21,8 @@
 //
 // mvo: taken from aptitude with a big _thankyou_ 
 
-#include "pkg_acqfile.h"
-
 #include "config.h"
+#include "pkg_acqfile.h"
 #include "i18n.h"
 
 #include <stdio.h>
--- synaptic/common/rcacheactor.cc
+++ synaptic/common/rcacheactor.cc
@@ -1,4 +1,5 @@
 
+#include "config.h"
 #include "rcacheactor.h"
 #include "rpackagelister.h"
 #include "i18n.h"
--- synaptic/common/rconfiguration.cc
+++ synaptic/common/rconfiguration.cc
@@ -34,6 +34,9 @@
 #include <apt-pkg/error.h>
 #include <apt-pkg/configuration.h>
 
+#include <unistd.h>
+#include <sys/types.h>
+
 #include "rconfiguration.h"
 
 #include "i18n.h"
--- synaptic/common/rpackagelistactor.cc
+++ synaptic/common/rpackagelistactor.cc
@@ -1,4 +1,5 @@
 
+#include "config.h"
 #include "rpackagelistactor.h"
 #include "rpackagelister.h"
 #include "i18n.h"
--- synaptic/common/rpackageview.cc
+++ synaptic/common/rpackageview.cc
@@ -22,6 +22,7 @@
  * USA
  */
 
+#include "config.h"
 #include <apt-pkg/pkgrecords.h>
 #include <apt-pkg/configuration.h>
 #include <rpackage.h>
--- synaptic/common/rpmindexcopy.cc
+++ synaptic/common/rpmindexcopy.cc
@@ -1,4 +1,5 @@
 
+#include "config.h"
 #include <apt-pkg/error.h>
 #include <apt-pkg/progress.h>
 #include <apt-pkg/strutl.h>
--- synaptic/common/rsources.cc
+++ synaptic/common/rsources.cc
@@ -23,6 +23,7 @@
  * USA
  */
 
+#include "config.h"
 #include <sys/stat.h>
 #include <dirent.h>
 
@@ -34,7 +35,6 @@
 #include <apt-pkg/error.h>
 #include <algorithm>
 #include <fstream>
-#include "config.h"
 #include "i18n.h"
 
 SourcesList::~SourcesList()
--- synaptic/common/rswig.cc
+++ synaptic/common/rswig.cc
@@ -1,5 +1,6 @@
 // 
 
+#include "config.h"
 #include "rswig.h"
 
 bool RInitSystem()
--- synaptic/common/rtagcollbuilder.cc
+++ synaptic/common/rtagcollbuilder.cc
@@ -20,4 +20,5 @@
 
 //#pragma implementation
 
+#include "config.h"
 #include "rtagcollbuilder.h"
--- synaptic/common/rtagcollfilter.cc
+++ synaptic/common/rtagcollfilter.cc
@@ -20,4 +20,5 @@
 
 //#pragma implementation
 
+#include "config.h"
 #include "rtagcollfilter.h"
--- synaptic/common/ruserdialog.cc
+++ synaptic/common/ruserdialog.cc
@@ -21,6 +21,7 @@
  * USA
  */
 
+#include "config.h"
 #include <apt-pkg/error.h>
 
 #include <ruserdialog.h>
--- synaptic/common/sections_trans.cc
+++ synaptic/common/sections_trans.cc
@@ -3,6 +3,7 @@
  *  
  */
 
+#include "config.h"
 #include <libintl.h>
 
 #include "sections_trans.h"
--- synaptic/configure.in
+++ synaptic/configure.in
@@ -13,6 +13,7 @@ AC_HEADER_STDC
 AC_PROG_CXX
 AC_PROG_MAKE_SET
 AC_PROG_RANLIB
+AC_SYS_LARGEFILE
 
 dnl i18n
 ALL_LINGUAS="ar pt_BR es fr gl de tr zh_CN zh_TW ru nl ja be it pl cs hu sr sr@Latn da he ca ko bg no nb sl pt lt el xh mk ro hi sv fi th"
--- synaptic/gtk/gsynaptic.cc
+++ synaptic/gtk/gsynaptic.cc
@@ -20,10 +20,12 @@
  * USA
  */
 
-#include <iostream>
-
 #include "config.h"
 
+#include <iostream>
+#include <cstdlib>
+#include <cstring>
+
 #include "i18n.h"
 
 #include "rconfiguration.h"
--- synaptic/gtk/gtkpkglist.cc
+++ synaptic/gtk/gtkpkglist.cc
@@ -17,6 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include "config.h"
 #include <algorithm>
 #include <gtk/gtk.h>
 #include <apt-pkg/configuration.h>
--- synaptic/gtk/rgaboutpanel.cc
+++ synaptic/gtk/rgaboutpanel.cc
@@ -22,8 +22,9 @@
  * USA
  */
 
-#include <cassert>
 #include "config.h"
+#include <cassert>
+#include <cstring>
 #include "rgaboutpanel.h"
 #include "i18n.h"
 
@@ -58,7 +59,7 @@ void RGAboutPanel::creditsClicked(GtkWidget *self, void *data)
    RGAboutPanel *me = (RGAboutPanel *) data;
 
    if (me->credits == NULL) {
-      me->credits = new RGCreditsPanel::RGCreditsPanel(me);
+      me->credits = new RGCreditsPanel(me);
    }
    me->credits->setTitle(_("Credits"));
 
--- synaptic/gtk/rgfiltermanager.cc
+++ synaptic/gtk/rgfiltermanager.cc
@@ -23,9 +23,10 @@
  */
 
 
+#include "config.h"
 #include <stdio.h>
 #include <cassert>
-#include "config.h"
+#include <cstring>
 #include "rpackageview.h"
 #include "rgfiltermanager.h"
 
--- synaptic/gtk/rgfindwindow.cc
+++ synaptic/gtk/rgfindwindow.cc
@@ -23,6 +23,7 @@
 #include "config.h"
 
 #include <cassert>
+#include <cstring>
 
 #include "rgfindwindow.h"
 #include "rgmisc.h"
--- synaptic/gtk/rggladewindow.cc
+++ synaptic/gtk/rggladewindow.cc
@@ -21,9 +21,9 @@
  */
 
 
+#include "config.h"
 #include <apt-pkg/fileutl.h>
 #include <cassert>
-#include "config.h"
 #include "i18n.h"
 #include "rggladewindow.h"
 
--- synaptic/gtk/rgiconlegend.cc
+++ synaptic/gtk/rgiconlegend.cc
@@ -20,8 +20,8 @@
  * USA
  */
 
-#include <cassert>
 #include "config.h"
+#include <cassert>
 #include "rgiconlegend.h"
 #include "rgmisc.h"
 
--- synaptic/gtk/rginstallprogress.cc
+++ synaptic/gtk/rginstallprogress.cc
@@ -32,6 +32,7 @@
 
 #include <unistd.h>
 #include <stdio.h>
+#include <cstring>
 #include <cstdlib>
 
 #include "i18n.h"
--- synaptic/gtk/rglogview.cc
+++ synaptic/gtk/rglogview.cc
@@ -20,9 +20,10 @@
  * USA
  */
 
+#include "config.h"
 #include <cassert>
+#include <cstring>
 #include <map>
-#include "config.h"
 #include "rglogview.h"
 #include "rgmisc.h"
 
--- synaptic/gtk/rgmisc.cc
+++ synaptic/gtk/rgmisc.cc
@@ -20,10 +20,12 @@
  * USA
  */
 
+#include "config.h"
 #include <X11/Xlib.h>
 #include <gdk/gdkx.h>
 #include <gtk/gtk.h>
-#include <string>
+#include <cstring>
+#include <cstdlib>
 #include "i18n.h"
 #include "rgmisc.h"
 #include <stdio.h>
--- synaptic/gtk/rgpreferenceswindow.cc
+++ synaptic/gtk/rgpreferenceswindow.cc
@@ -28,6 +28,8 @@
 #include <apt-pkg/configuration.h>
 #include <gtk/gtk.h>
 #include <cassert>
+#include <stdlib.h>
+#include <cstring>
 
 #include "rconfiguration.h"
 #include "rgpreferenceswindow.h"
--- synaptic/gtk/rgrepositorywin.cc
+++ synaptic/gtk/rgrepositorywin.cc
@@ -23,6 +23,7 @@
  * USA
  */
 
+#include "config.h"
 #include <apt-pkg/error.h>
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/sourcelist.h>
@@ -31,7 +32,6 @@
 #include "rgrepositorywin.h"
 #include "rguserdialog.h"
 #include "rgmisc.h"
-#include "config.h"
 #include "i18n.h"
 
 #if HAVE_RPM
--- synaptic/gtk/rgsetoptwindow.cc
+++ synaptic/gtk/rgsetoptwindow.cc
@@ -20,11 +20,9 @@
  * USA
  */
 
-#include <apt-pkg/configuration.h>
-
 #include "config.h"
+#include <apt-pkg/configuration.h>
 #include "i18n.h"
-
 #include "rgsetoptwindow.h"
 
 void RGSetOptWindow::DoApply(GtkWindow *widget, void *data)
--- synaptic/gtk/rgslideshow.cc
+++ synaptic/gtk/rgslideshow.cc
@@ -1,3 +1,4 @@
+#include "config.h"
 #include <sys/types.h>
 #include <dirent.h>
 
@@ -6,6 +7,7 @@
 #include <algorithm>
 #include <string>
 #include <vector>
+#include <algorithm>
 
 #include "rgslideshow.h"
 
--- synaptic/gtk/rgtaskswin.cc
+++ synaptic/gtk/rgtaskswin.cc
@@ -20,10 +20,10 @@
  * USA
  */
 
+#include "config.h"
 #include <gtk/gtk.h>
 #include <cassert>
 #include <list>
-#include "config.h"
 #include "rgtaskswin.h"
 #include "rgmainwindow.h"
 #include "rguserdialog.h"
--- synaptic/gtk/rgterminstallprogress.cc
+++ synaptic/gtk/rgterminstallprogress.cc
@@ -37,6 +37,8 @@
 #include <X11/Xlib.h>
 #include <iostream>
 #include <cerrno>
+#include <cstring>
+#include <cstdlib>
 #include <unistd.h>
 #include <sys/wait.h>
 #include <sys/types.h>
--- synaptic/gtk/rguserdialog.cc
+++ synaptic/gtk/rguserdialog.cc
@@ -22,6 +22,7 @@
  * USA
  */
 
+#include "config.h"
 #include <apt-pkg/error.h>
 #include <apt-pkg/fileutl.h>
 
--- synaptic/gtk/rgvendorswindow.cc
+++ synaptic/gtk/rgvendorswindow.cc
@@ -1,8 +1,8 @@
 /* $Id: rgvendorswindow.cc,v 1.4 2002/12/06 08:16:40 mvogt Exp $ */
+#include "config.h"
 #include <cassert>
 #include "rgvendorswindow.h"
 #include "rgrepositorywin.h"
-#include "config.h"
 #include "i18n.h"
 
 RGVendorsEditor::RGVendorsEditor(RGWindow *parent, SourcesList &_lst)
--- synaptic/gtk/rgwindow.cc
+++ synaptic/gtk/rgwindow.cc
@@ -21,8 +21,8 @@
  */
 
 
-#include <apt-pkg/fileutl.h>
 #include "config.h"
+#include <apt-pkg/fileutl.h>
 #include "i18n.h"
 #include "rgwindow.h"
 
