diff --git a/synaptic/common/rpackage.cc b/synaptic/common/rpackage.cc
index 90d94c0..c9ac0e2 100644
--- a/synaptic/common/rpackage.cc
+++ b/synaptic/common/rpackage.cc
@@ -748,7 +748,7 @@ void RPackage::setKeep()
 
 void RPackage::setInstall()
 {
-   _depcache->MarkInstall(*_package, true);
+   _depcache->MarkInstall(*_package, pkgDepCache::AutoMarkFlag::Manual, true);
    pkgDepCache::StateCache & State = (*_depcache)[*_package];
 
    // FIXME: can't we get rid of it here?
diff --git a/synaptic/common/rpackagelister.cc b/synaptic/common/rpackagelister.cc
index 85fa033..e45ffd1 100644
--- a/synaptic/common/rpackagelister.cc
+++ b/synaptic/common/rpackagelister.cc
@@ -1047,10 +1047,10 @@ void RPackageLister::restoreState(RPackageLister::pkgState &state)
 
       if (oldflags != flags) {
          if (oldflags & RPackage::FReInstall) {
-            deps->MarkInstall(*(pkg->package()), true);
+            deps->MarkInstall(*(pkg->package()), pkgDepCache::AutoMarkFlag::DontChange, true);
             deps->SetReInstall(*(pkg->package()), false);
          } else if (oldflags & RPackage::FInstall) {
-            deps->MarkInstall(*(pkg->package()), true);
+            deps->MarkInstall(*(pkg->package()), pkgDepCache::AutoMarkFlag::Manual, true);
          } else if (oldflags & RPackage::FRemove) {
             deps->MarkDelete(*(pkg->package()), oldflags & RPackage::FPurge);
          } else if (oldflags & RPackage::FKeep) {
@@ -1803,9 +1803,9 @@ bool RPackageLister::readSelections(istream &in)
 		  if(_config->FindB("Volatile::SetSelectionDoReInstall",false))
 		     Cache.SetReInstall(Pkg, true);
 		  if(_config->FindB("Volatile::SetSelectionsNoFix",false))
-		     Cache.MarkInstall(Pkg, false);
+		     Cache.MarkInstall(Pkg, pkgDepCache::AutoMarkFlag::Manual, false);
 		  else
-		     Cache.MarkInstall(Pkg, true);
+		     Cache.MarkInstall(Pkg, pkgDepCache::AutoMarkFlag::Manual, true);
                   break;
 
                case ACTION_UNINSTALL:
