Index: gtk/rgchangeswindow.h
===================================================================
--- gtk/rgchangeswindow.h	(revision 1323)
+++ gtk/rgchangeswindow.h	(working copy)
@@ -31,8 +31,6 @@
 class RPackageLister;
 
 class RGChangesWindow:public RGGladeWindow {
-   bool _confirmed;
-
    enum {
       PKG_COLUMN,
       N_COLUMNS
@@ -41,19 +39,16 @@
    GtkWidget *_tree;
    GtkTreeStore *_treeStore;
 
-   static void clickedOk(GtkWidget *w, void *data);
-   static void clickedCancel(GtkWidget *w, void *data);
-
  public:
    RGChangesWindow(RGWindow *win);
 
-   bool showAndConfirm(RPackageLister *lister,
-                       vector<RPackage *> &kept,
-                       vector<RPackage *> &toInstall,
-                       vector<RPackage *> &toReInstall,
-                       vector<RPackage *> &toUpgrade,
-                       vector<RPackage *> &toRemove,
-                       vector<RPackage *> &toDowngrade);
+   void confirm(RPackageLister *lister,
+		vector<RPackage *> &kept,
+		vector<RPackage *> &toInstall,
+		vector<RPackage *> &toReInstall,
+		vector<RPackage *> &toUpgrade,
+		vector<RPackage *> &toRemove,
+		vector<RPackage *> &toDowngrade);
 };
 
 #endif
Index: gtk/rgmainwindow.cc
===================================================================
--- gtk/rgmainwindow.cc	(revision 1323)
+++ gtk/rgmainwindow.cc	(working copy)
@@ -521,11 +521,11 @@
    if (ask && _lister->getStateChanges(state, toKeep, toInstall, toReInstall,
                                            toUpgrade, toRemove, toDowngrade,
                                            exclude)) {
-      RGChangesWindow *chng;
-      // show a summary of what's gonna happen
-      chng = new RGChangesWindow(this);
-      if (!chng->showAndConfirm(_lister, toKeep, toInstall, toReInstall,
-                                toUpgrade, toRemove, toDowngrade)) {
+      RGChangesWindow changes(this);
+      changes.confirm(_lister, toKeep, toInstall, toReInstall,
+		   toUpgrade, toRemove, toDowngrade);
+      int res = gtk_dialog_run(GTK_DIALOG(changes.window()));
+      if( res != GTK_RESPONSE_OK) {
          // canceled operation
          _lister->restoreState(state);
 	 // if a operation was canceled, we discard all errors from this
@@ -533,7 +533,6 @@
 	 _error->Discard();
          changed = false;
       }
-      delete chng;
    }
 
    return changed;
Index: gtk/window_summary.glade
===================================================================
--- gtk/window_summary.glade	(revision 1323)
+++ gtk/window_summary.glade	(working copy)
@@ -2,17 +2,21 @@
 <!DOCTYPE glade-interface SYSTEM "http://glade.gnome.org/glade-2.0.dtd">
 
 <glade-interface>
-<requires lib="gnome"/>
 
 <widget class="GtkDialog" id="window_summary">
   <property name="border_width">6</property>
   <property name="title" translatable="yes"></property>
   <property name="type">GTK_WINDOW_TOPLEVEL</property>
   <property name="window_position">GTK_WIN_POS_NONE</property>
-  <property name="modal">False</property>
+  <property name="modal">True</property>
   <property name="resizable">True</property>
   <property name="destroy_with_parent">False</property>
   <property name="icon">synaptic_mini.xpm</property>
+  <property name="decorated">True</property>
+  <property name="skip_taskbar_hint">True</property>
+  <property name="skip_pager_hint">False</property>
+  <property name="type_hint">GDK_WINDOW_TYPE_HINT_DIALOG</property>
+  <property name="gravity">GDK_GRAVITY_NORTH_WEST</property>
   <property name="has_separator">False</property>
 
   <child internal-child="vbox">
@@ -35,6 +39,7 @@
 	      <property name="label">gtk-cancel</property>
 	      <property name="use_stock">True</property>
 	      <property name="relief">GTK_RELIEF_NORMAL</property>
+	      <property name="focus_on_click">True</property>
 	      <property name="response_id">-6</property>
 	      <signal name="clicked" handler="on_cancelbutton1_clicked" last_modification_time="Thu, 20 Nov 2003 13:50:29 GMT"/>
 	    </widget>
@@ -49,6 +54,7 @@
 	      <property name="label">gtk-apply</property>
 	      <property name="use_stock">True</property>
 	      <property name="relief">GTK_RELIEF_NORMAL</property>
+	      <property name="focus_on_click">True</property>
 	      <property name="response_id">-10</property>
 	      <signal name="clicked" handler="on_button_execute_clicked" last_modification_time="Thu, 20 Nov 2003 13:50:19 GMT"/>
 	    </widget>
@@ -362,6 +368,7 @@
 			      <property name="label" translatable="yes">_Show Details</property>
 			      <property name="use_underline">True</property>
 			      <property name="relief">GTK_RELIEF_NORMAL</property>
+			      <property name="focus_on_click">True</property>
 			      <property name="active">False</property>
 			      <property name="inconsistent">False</property>
 			    </widget>
@@ -422,6 +429,7 @@
 			  <property name="label" translatable="yes">Download package files only</property>
 			  <property name="use_underline">True</property>
 			  <property name="relief">GTK_RELIEF_NORMAL</property>
+			  <property name="focus_on_click">True</property>
 			  <property name="active">False</property>
 			  <property name="inconsistent">False</property>
 			  <property name="draw_indicator">True</property>
@@ -440,6 +448,7 @@
 			  <property name="label" translatable="yes">Verify package signatures</property>
 			  <property name="use_underline">True</property>
 			  <property name="relief">GTK_RELIEF_NORMAL</property>
+			  <property name="focus_on_click">True</property>
 			  <property name="active">False</property>
 			  <property name="inconsistent">False</property>
 			  <property name="draw_indicator">True</property>
Index: gtk/rgchangeswindow.cc
===================================================================
--- gtk/rgchangeswindow.cc	(revision 1323)
+++ gtk/rgchangeswindow.cc	(working copy)
@@ -38,23 +38,7 @@
 
 #include "i18n.h"
 
-void RGChangesWindow::clickedOk(GtkWidget *self, void *data)
-{
-   RGChangesWindow *me = (RGChangesWindow *) data;
-   me->_confirmed = true;
-   gtk_main_quit();
-}
 
-
-void RGChangesWindow::clickedCancel(GtkWidget *self, void *data)
-{
-   //cout << "clickedCancel()" << endl;
-   RGChangesWindow *me = (RGChangesWindow *) data;
-   me->_confirmed = false;
-   gtk_main_quit();
-}
-
-
 RGChangesWindow::RGChangesWindow(RGWindow *wwin)
 : RGGladeWindow(wwin, "changes")
 {
@@ -72,23 +56,11 @@
    /* Add the column to the view. */
    gtk_tree_view_append_column(GTK_TREE_VIEW(_tree), column);
    gtk_widget_show(_tree);
-
-   glade_xml_signal_connect_data(_gladeXML,
-                                 "on_proceed_clicked",
-                                 G_CALLBACK(clickedOk), this);
-   glade_xml_signal_connect_data(_gladeXML,
-                                 "on_cancel_clicked",
-                                 G_CALLBACK(clickedCancel), this);
-   glade_xml_signal_connect_data(_gladeXML,
-                                 "on_window_changes_close",
-                                 G_CALLBACK(clickedCancel), this);
-
-   skipTaskbar(true);
 }
 
 
 
-bool RGChangesWindow::showAndConfirm(RPackageLister *lister,
+void RGChangesWindow::confirm(RPackageLister *lister,
                                      vector<RPackage *> &kept,
                                      vector<RPackage *> &toInstall,
                                      vector<RPackage *> &toReInstall,
@@ -173,10 +145,4 @@
    }
 
    gtk_tree_view_expand_all(GTK_TREE_VIEW(_tree));
-
-   show();
-   gtk_window_set_modal(GTK_WINDOW(_win), TRUE);
-   gtk_main();
-
-   return _confirmed;
 }
Index: gtk/window_changes.glade
===================================================================
--- gtk/window_changes.glade	(revision 1323)
+++ gtk/window_changes.glade	(working copy)
@@ -8,15 +8,19 @@
   <property name="visible">True</property>
   <property name="title" translatable="yes"></property>
   <property name="type">GTK_WINDOW_TOPLEVEL</property>
-  <property name="window_position">GTK_WIN_POS_NONE</property>
+  <property name="window_position">GTK_WIN_POS_CENTER_ON_PARENT</property>
   <property name="modal">True</property>
   <property name="default_width">450</property>
   <property name="default_height">350</property>
   <property name="resizable">True</property>
   <property name="destroy_with_parent">False</property>
   <property name="icon">synaptic_mini.xpm</property>
+  <property name="decorated">True</property>
+  <property name="skip_taskbar_hint">True</property>
+  <property name="skip_pager_hint">False</property>
+  <property name="type_hint">GDK_WINDOW_TYPE_HINT_DIALOG</property>
+  <property name="gravity">GDK_GRAVITY_NORTH_WEST</property>
   <property name="has_separator">False</property>
-  <signal name="hide" handler="on_window_changes_close" last_modification_time="Sun, 04 Jan 2004 17:45:45 GMT"/>
 
   <child internal-child="vbox">
     <widget class="GtkVBox" id="dialog-vbox1">
@@ -37,8 +41,8 @@
 	      <property name="label">gtk-cancel</property>
 	      <property name="use_stock">True</property>
 	      <property name="relief">GTK_RELIEF_NORMAL</property>
+	      <property name="focus_on_click">True</property>
 	      <property name="response_id">-6</property>
-	      <signal name="clicked" handler="on_cancel_clicked" last_modification_time="Tue, 12 Aug 2003 10:03:15 GMT"/>
 	    </widget>
 	  </child>
 
@@ -48,8 +52,8 @@
 	      <property name="can_default">True</property>
 	      <property name="can_focus">True</property>
 	      <property name="relief">GTK_RELIEF_NORMAL</property>
+	      <property name="focus_on_click">True</property>
 	      <property name="response_id">-5</property>
-	      <signal name="clicked" handler="on_proceed_clicked" last_modification_time="Tue, 12 Aug 2003 10:02:07 GMT"/>
 
 	      <child>
 		<widget class="GtkAlignment" id="alignment1">
@@ -58,6 +62,10 @@
 		  <property name="yalign">0.5</property>
 		  <property name="xscale">0</property>
 		  <property name="yscale">0</property>
+		  <property name="top_padding">0</property>
+		  <property name="bottom_padding">0</property>
+		  <property name="left_padding">0</property>
+		  <property name="right_padding">0</property>
 
 		  <child>
 		    <widget class="GtkHBox" id="hbox2">
