--- synaptic.orig/common/rsources.cc
+++ synaptic/common/rsources.cc
@@ -326,10 +326,10 @@ bool SourcesList::UpdateSources()
                S += "[" + (*it)->VendorID + "] ";
 
             S += (*it)->URI + " ";
-            S += (*it)->Dist + " ";
+            S += (*it)->Dist;
 
             for (unsigned int J = 0; J < (*it)->NumSections; J++)
-               S += (*it)->Sections[J] + " ";
+               S += " " + (*it)->Sections[J];
          }
          ofs << S << endl;
       }
