--- synaptic.orig/common/rsources.cc
+++ synaptic/common/rsources.cc
@@ -433,7 +433,11 @@ bool SourcesList::ReadVendors()
 {
    Configuration Cnf;
 
-   string CnfFile = _config->FindFile("Dir::Etc::vendorlist");
+   string CnfFile = _config->FindDir("Dir::Etc::vendorparts");
+   if (FileExists(CnfFile) == true)
+      if (ReadConfigDir(Cnf,CnfFile,true) == false)
+         return false;
+   CnfFile = _config->FindFile("Dir::Etc::vendorlist");
    if (FileExists(CnfFile) == true)
       if (ReadConfigFile(Cnf, CnfFile, true) == false)
          return false;
