--- synaptic.orig/common/rsources.cc
+++ synaptic/common/rsources.cc
@@ -326,12 +326,13 @@ bool SourcesList::UpdateSources()
                S += "[" + (*it)->VendorID + "] ";
 
             S += (*it)->URI + " ";
-            S += (*it)->Dist + " ";
+            S += (*it)->Dist;
 
             for (unsigned int J = 0; J < (*it)->NumSections; J++)
-               S += (*it)->Sections[J] + " ";
+               S += " " + (*it)->Sections[J];
          }
-         ofs << S << endl;
+         if ( ! S.empty() ) // we needn't to write empty line ...
+	     ofs << S << endl;
       }
       ofs.close();
    }
--- synaptic.orig/gtk/rgrepositorywin.cc
+++ synaptic/gtk/rgrepositorywin.cc
@@ -675,11 +675,13 @@ void RGRepositoryEditor::SelectionChanged(GtkTreeSelection *selection,
       gtk_entry_set_text(GTK_ENTRY(me->_entryDist), utf8(rec->Dist.c_str()));
       gtk_entry_set_text(GTK_ENTRY(me->_entrySect), "");
 
-      for (unsigned int I = 0; I < rec->NumSections; I++) {
+      for (unsigned int I = 0; I < (rec->NumSections - 1); I++) {
          gtk_entry_append_text(GTK_ENTRY(me->_entrySect),
                                utf8(rec->Sections[I].c_str()));
          gtk_entry_append_text(GTK_ENTRY(me->_entrySect), " ");
       }
+      gtk_entry_append_text(GTK_ENTRY(me->_entrySect),
+               utf8(rec->Sections[rec->NumSections - 1].c_str()));
    } else {
       //cout << "no selection" << endl;
       gtk_widget_set_sensitive(me->_editTable, FALSE);
