--- synaptic-0.57.7/configure.in.orig	2010-10-18 12:39:32 +0400
+++ synaptic-0.57.7/configure.in	2010-10-18 13:51:20 +0400
@@ -121,7 +121,7 @@
 
 DEB_HDRS=""
 AC_SUBST(DEB_HDRS)
-DEB_LIBS="-lapt-inst"
+DEB_LIBS=""
 AC_SUBST(DEB_LIBS)
 
 # launchpad integration
--- synaptic/common/rpackage.cc.orig
+++ synaptic/common/rpackage.cc
@@ -128,8 +128,7 @@ const char *RPackage::summary()
 
    pkgCache::VerIterator ver = (*_depcache)[*_package].CandidateVerIter(*_depcache);
    if (!ver.end()) {
-      pkgCache::DescIterator Desc = ver.TranslatedDescription();
-      pkgRecords::Parser & parser = _records->Lookup(Desc.FileList());
+      pkgRecords::Parser & parser = _records->Lookup(ver.FileList());
       _summary = parser.ShortDesc();
       return _summary.c_str();
    }
diff --git a/synaptic/common/rpackagelister.cc b/synaptic/common/rpackagelister.cc
index 85fa033..5e20ec6 100644
--- synaptic/common/rpackagelister.cc.orig
+++ synaptic/common/rpackagelister.cc
@@ -324,8 +324,12 @@ bool RPackageLister::openCache()
                              "Please report."), 2);
    }
 
+#ifdef HAVE_RPM
+   // be gentle and free memory
    if (_records)
       delete _records;
+#endif
+
    _records = new pkgRecords(*deps);
 
    if (_error->PendingError()) {
